21-RegisterForNavigation/src/PrismSample/ViewModels/MainMenuPageViewModel.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Prism.Commands;
+using Prism.Navigation;
+
+namespace PrismSample.ViewModels
+{
+    public class MainMenuPageViewModel
+    {
+        private readonly INavigationService _navigationService;
+
+        public DelegateCommand GoToBasicPageCommand { get; private set; }
+        public DelegateCommand GoToIdiomSpecificPageCommand { get; private set; }
+        public DelegateCommand GoToFullRegistratedPageCommand { get; private set; }
+        public DelegateCommand GoToPlatformSpecificPageCommand { get; private set; }
+        public DelegateCommand GoToCustomNameRegisteredPageCommand { get; private set; }
+
+        public MainMenuPageViewModel(INavigationService navigationService)
+        {
+            _navigationService = navigationService;
+
+            GoToBasicPageCommand = new DelegateCommand(GoToBasicPage);
+            GoToIdiomSpecificPageCommand = new DelegateCommand(GoToIdiomSpecificPage);
+            GoToFullRegistratedPageCommand = new DelegateCommand(GoToFullRegistratedPage);
+            GoToPlatformSpecificPageCommand = new DelegateCommand(GoToPlatformSpecificPage);
+            GoToCustomNameRegisteredPageCommand = new DelegateCommand(GoToCustomNameRegisteredPage);
+        }
```

Comment:
Replace this madness with a single NavigateCommand:

```cs
public DelegateCommand<string> NavigateCommand { get; }

public async void OnNavigateCommandExecuted(string uri) =>
    await _navigationService.NavigateAsync(uri);
```

Reply:
You are 100% right about this! In my defence, I was writing this in a train going back from hometown 😂